task/power/src/main.rs

```diff
@@ -349,14 +297,17 @@ fn get_state() -> PowerState {
 fn main() -> ! {
     let sensor = sensor_api::Sensor::from(SENSOR.get_task_id());
 
-    let mut controllers = controllers();
+    let i2c_task = I2C.get_task_id();
+
+    let mut devices: [Device; CONTROLLER_CONFIG.len()] =
+        array_init::array_init(|i| CONTROLLER_CONFIG[i].get_device(i2c_task));
```

Comment:
Do we need all the devices on the stack, or could we create them on demand in the loop below? Would only be a win if the creation is very cheap (in both text and runtime), but at a glance it looks like it might be, and would let us buy back a little chunk of stack (although maybe not enough to bother).

Reply:
You might also consider using `lib/mutable-statics` to put this in static memory, where its usage is more cleanly accounted for. (This would also require some dancing with either a default value or MaybeUninit; I should add an `array_init`-style version of the mutable static types....)

Reply:
Some of the devices have local caches of values, so we do need them to be persistent.

(only _some_ of them, but separating them into persistent vs transient would be a bit of a mess)